pkg/scheduler/ugm/queue_tracker.go

```diff
@@ -201,11 +274,75 @@ func (qt *QueueTracker) getResourceUsageDAOInfo(parentQueuePath string) *dao.Res
 	for app := range qt.runningApplications {
 		usage.RunningApplications = append(usage.RunningApplications, app)
 	}
-	usage.MaxResources = qt.maxResourceUsage
+	usage.MaxResources = qt.maxResources
 	usage.MaxApplications = qt.maxRunningApps
 	for _, cqt := range qt.childQueueTrackers {
 		childUsage := cqt.getResourceUsageDAOInfo(fullQueuePath)
 		usage.Children = append(usage.Children, childUsage)
 	}
 	return usage
 }
+
+// IsQueuePathTrackedCompletely Traverse queue path upto the end queue through its linkage
+// to confirm entire queuePath has been tracked completely or not
+func (qt *QueueTracker) IsQueuePathTrackedCompletely(queuePath string) bool {
+	if queuePath == configs.RootQueue || queuePath == qt.queueName {
+		return true
+	}
+	childQueuePath, immediateChildQueueName := getChildQueuePath(queuePath)
+	if immediateChildQueueName != "" {
+		if childUt, ok := qt.childQueueTrackers[immediateChildQueueName]; ok {
+			return childUt.IsQueuePathTrackedCompletely(childQueuePath)
+		}
+	}
+	return false
+}
+
+// IsUnlinkRequired Traverse queue path upto the leaf queue and decide whether
+// linkage needs to be removed or not based on the running applications.
+// If there are any running applications in end leaf queue, we should remove the linkage between
+// the leaf and its parent queue using UnlinkQT method. Otherwise, we should leave as it is.
+func (qt *QueueTracker) IsUnlinkRequired(queuePath string) bool {
+	childQueuePath, immediateChildQueueName := getChildQueuePath(queuePath)
+	if immediateChildQueueName != "" {
+		if childUt, ok := qt.childQueueTrackers[immediateChildQueueName]; ok {
+			return childUt.IsUnlinkRequired(childQueuePath)
+		}
+	}
+	if queuePath == configs.RootQueue || queuePath == qt.queueName {
+		if len(qt.runningApplications) == 0 {
+			log.Log(log.SchedUGM).Debug("Is Unlink Required?",
+				zap.String("queue path", queuePath),
+				zap.Int("no. of applications", len(qt.runningApplications)))
+			return true
+		}
+	}
+	return false
+}
+
+// UnlinkQT Traverse queue path upto the end queue. If end queue has any more child queue trackers,
+// then goes upto each child queue and removes the linkage with its immediate parent
+func (qt *QueueTracker) UnlinkQT(queuePath string) bool {
```

Comment:
Parent not needed if we use recursive checks.

root.parent.**child**.leaf
 -> slice of [root, parent, child, leaf]
```
manager -> user.rootQT.unlinkChild(path []slice)
root.unlinkChild([parent, child, leaf])
  root -> find parent
parent.unlinkChild([child.leaf])
   parent -> find child
child.unlinkQT([leaf])
if slice.len ==1 leaf can be removed?
  delete(qt.childQueueTrackers, leaf.queueName)
```